libs/platform/ebpf_platform.h

```diff
@@ -217,47 +215,45 @@ extern "C"
      * @retval EBPF_ERROR_ARITHMETIC_OVERFLOW Addition overflowed or underflowed.
      */
     ebpf_result_t
-    ebpf_safe_size_t_subtract(size_t minuend, size_t subtrahend, size_t* result);
+    ebpf_safe_size_t_subtract(size_t minuend, size_t subtrahend, _Out_ size_t* result);
 
     /**
      * @brief Create an instance of a lock.
-     * @param[in] lock Pointer to memory location that will contain the lock.
+     * @param[out] lock Pointer to memory location that will contain the lock.
      */
     void
-    ebpf_lock_create(ebpf_lock_t* lock);
+    ebpf_lock_create(_Out_ ebpf_lock_t* lock);
 
     /**
      * @brief Destroy an instance of a lock.
      * @param[in] lock Pointer to memory location that contains the lock.
      */
     void
-    ebpf_lock_destroy(ebpf_lock_t* lock);
+    ebpf_lock_destroy(_In_ ebpf_lock_t* lock);
 
     /**
      * @brief Acquire exclusive access to the lock.
      * @param[in] lock Pointer to memory location that contains the lock.
-     * @param[out] state Pointer to memory location that contains state that
-     *    needs to be passed to ebpf_lock_unlock.
+     * @returns - The previous lock_state required for unlock.
      */
-    void
-    ebpf_lock_lock(ebpf_lock_t* lock, ebpf_lock_state_t* state);
+    _Requires_lock_not_held_(*lock) _Acquires_lock_(*lock) _IRQL_requires_max_(DISPATCH_LEVEL) _IRQL_saves_
+        _IRQL_raises_(DISPATCH_LEVEL) ebpf_lock_state_t ebpf_lock_lock(_In_ ebpf_lock_t* lock);
```

Comment:
Not needed in this PR, but for future reference ebpf_lock_lock is a redundant name.  either ebpf_lock_acquire or just ebpf_lock would be better.